compiler/expressions.cpp

```diff
@@ -2062,39 +2066,11 @@ SC3ExpressionParser::hier1(value* lval1)
             }
             if (gTypes.find(sym->x.tags.index)->isEnumStruct())
                 error(117);
-            /* if the array index is a field from an enumeration, get the tag name
-             * from the field and save the size of the field too. Otherwise, the
-             * tag is the one from the array symbol.
-             */
-            if (lval2.ident == iCONSTEXPR && lval2.sym && lval2.sym->parent() &&
-                lval2.sym->dim.array.length > 0 && sym->dim.array.level == 0)
-            {
-                lval1->tag = lval2.sym->x.tags.index;
-                lval1->constval = lval2.sym->dim.array.length;
-                if (lval2.tag == sym->x.tags.index && lval1->constval > 1 && matchtoken('[')) {
-                    /* an array indexed with an enumeration field may be considered a sub-array */
-                    lexpush();
-                    lvalue = FALSE; /* for now, a iREFARRAY is no lvalue */
-                    lval1->ident = iREFARRAY;
-                    /* initialize a dummy symbol, which is a copy of the current symbol,
-                     * but with an adjusted index tag
-                     */
-                    assert(sym != NULL);
-                    new (&dummysymbol) symbol(*sym);
-                    /* get the tag of the root of the enumeration */
-                    assert(lval2.sym != NULL);
-                    dummysymbol.x.tags.index = lval2.sym->x.tags.field;
-                    dummysymbol.dim.array.length = lval2.sym->dim.array.length;
-                    cursym = &dummysymbol;
-                    /* recurse */
-                    goto restart;
-                }
-            } else {
-                assert(sym != NULL);
-                if (cursym != &dummysymbol)
-                    lval1->tag = sym->tag;
-                lval1->constval = 0;
-            }
+
+            assert(sym != NULL);
+            if (cursym != &dummysymbol)
+                lval1->tag = sym->tag;

```

Comment:
This is dead now and `dummysymbol` is never used.

Reply:
Thanks, it gives me great joy to remove this declaration.